lib/search/index.ts

```diff
@@ -229,6 +257,11 @@ export const middleware: S.Middleware = (store) => {
     return [...pinnedMatches.values(), ...matches.values()];
   };
 
+  searchNotes = (args, maxResults) =>
+    runSearch(args, maxResults)
+      .map((noteId) => [noteId, store.getState().data.notes.get(noteId)])
+      .filter(([, a]) => 'undefined' !== typeof a) as [T.EntityId, T.Note][];
```

Comment:
I probably wrote this filtering line but we don't need it because we can have `runSearch` perform the check. this is faster and also avoids the case where we request a max number of results and get those, but then prune the list to return fewer than the maximum

```ts
const storeNotes = store.getState().data.notes;
…
const note = notes.get(noteId);
if (!note || !storeNotes.has(noteId)) {
	continue;
}
```

Reply:
Hmm I did this and it seems to work but I've made TypeScript sad. https://github.com/Automattic/simplenote-electron/pull/2286/commits/d850158639f17992901e8ab2b645aa8df3f9b332